backend/books/views.py

```diff
@@ -174,21 +183,43 @@ def create(self, request, *args, **kwargs):
         setDefaultImage = str2bool(data.get('setDefaultImage'))
 
         # Get and Create the Image
-        if ((request.FILES.get('image') is not None) or setDefaultImage):
+        if self.handle_image(request, setDefaultImage):
             url = self.bookimage_get_and_create(request, res.get('isbn_13'), setDefaultImage)
             res['image_url'] = url
 
         headers = self.get_success_headers(serializer.data)
         return Response(res, status=status.HTTP_201_CREATED, headers=headers)
 
+    def convert_zero_to_null(self, data):
+        possible_zero_fields = ['pageCount', 'width', 'height', 'thickness']
+        for possible_zero_field in possible_zero_fields:
+            v = data.get(possible_zero_field, None)
+            if v == '0' or v == 0:
+                data[possible_zero_field] = None
+
+        return data
+    
+    def handle_image(self, request, setDefaultImage):
+        has_image_bytes = self.has_image_bytes(request)
+        has_image_url = self.has_image_url(request)
+        return has_image_bytes or has_image_url or setDefaultImage
+    
+    def has_image_bytes(self, request):
+        return request.FILES.get('image_bytes', None) is not None
+
+    def has_image_url(self, request):
+        return request.data.get('image_url', None) is not None
+
     def bookimage_get_and_create(self, request, isbn_13, setDefaultImage):
         book = Book.objects.filter(isbn_13=isbn_13)
 
         # This creates an image in static and sends a file
         if setDefaultImage:
             url = self.isbn_toolbox.get_default_image_url()
-        else:
+        elif self.has_image_bytes(request):
             url = self.isbn_toolbox.commit_image_raw_bytes(request, book[0].id, isbn_13)
+        elif self.has_image_url(request):
+            url = self.isbn_toolbox.commit_image_url(request, book[0].id, isbn_13)
 
         obj, created = BookImage.objects.get_or_create(
             book_id=book[0].id,
```

Comment:
If none of these conditions are met, then `url` would be undefined. So either, why will one of these conditions always guaranteed to be met, or need to add some type of `else`

Reply:
line 186 `if self.handle_image(request, setDefaultImage):`